core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java

```diff
@@ -51,8 +51,8 @@
 	public static final boolean SERIALIZABLE_VALIDATION_DEFAULT = true;
 	public static final boolean ECLIPSE_RDF4J_SHACL_EXTENSIONS_DEFAULT = false;
 	public static final boolean DASH_DATA_SHAPES_DEFAULT = false;
-	public final static long VALIDATION_RESULTS_LIMIT_TOTAL_DEFAULT = -1;
-	public final static long VALIDATION_RESULTS_LIMIT_PER_CONSTRAINT_DEFAULT = -1;
+	public final static long VALIDATION_RESULTS_LIMIT_TOTAL_DEFAULT = 1_000_000;
+	public final static long VALIDATION_RESULTS_LIMIT_PER_CONSTRAINT_DEFAULT = 1_000;
```

Comment:
You guys think that these are reasonable limits?

Reply:
I'm not sure - if you had to do a "back of the envelope" calculation of how much memory such a truncated report would consume at most - any clue? Put another way: how large is a single validation result?

Reply:
I can give it a shot.

Reply:
Fwiw I'm fine with putting these limits in if you're confident these are reasonable - I just have no data on which to decide for myself how big 1 million validation results can get :)

Reply:
1 000 000 violations needs around 700 MB of RAM (eg. setValidationResultsLimitTotal(0) saves 700 MB of RAM).